src/dftbp/dftbplus/mainapi.F90

```diff
@@ -183,6 +183,39 @@ subroutine getGrossCharges(env, main, atomCharges)
 
   end subroutine getGrossCharges
 
+  !> get the CM5 charges
+  subroutine getCM5Charges(env, main, atomCharges)
+
+    !> instance
+    type(TEnvironment), intent(inout) :: env
+
+    !> Instance
+    type(TDftbPlusMain), intent(inout) :: main
+
+    !> resulting charges
+    real(dp), intent(out) :: atomCharges(:)
+
+    !> number of neighbours for all atoms
+    integer, allocatable :: nNeigh(:)
+    
+    !> handle the case that CM5 was not added in the input
+    if (.not. allocated(main%cm5Cont)) then
+      call error("CM5 analysis has not been carried out.")
+    end if
+
+    call recalcGeometry(env, main)
+    if (.not. allocated(main%cm5Cont%cm5)) then
+      call error("CM5 could not be calculated.")
+    end if
+    atomCharges(:) = sum(main%q0(:, :, 1) - main%qOutput(:, :, 1), dim=1) + main%cm5Cont%cm5
+
+    !> Pass to the charges of the excited state if relevant
+    if (main%isLinResp) then
+      atomCharges(:) = atomCharges(:) + main%dQAtomEx(:)
```

Comment:
I am wondering, why you add `main%dQAtomEx` here?

Reply:
That's a copy/paste mistake from the gross_atom_charges API call. Are there CM5 charges for the exited state available or should we just omit this?

Reply:
Not at the moment (but as they are additive, should be simple to implement).